assets/src/edit-story/components/panels/textStyle/padding.js

```diff
@@ -56,45 +59,103 @@ function PaddingControls({ selectedElements, pushUpdateForObject }) {
   const lockPadding =
     padding.locked === MULTIPLE_VALUE ? false : padding.locked;
 
-  const handleChange = useCallback(
-    (newPadding) => {
-      let update = newPadding;
-      if (lockPadding) {
-        const commonPadding =
-          newPadding.horizontal !== undefined
-            ? newPadding.horizontal
-            : newPadding.vertical;
-        update = {
-          horizontal: commonPadding,
-          vertical: commonPadding,
-          locked: true,
-        };
-      }
-      pushUpdateForObject('padding', update, DEFAULT_PADDING);
-    },
-    [pushUpdateForObject, lockPadding]
-  );
-
   return (
     <Row>
       <Label>{__('Padding', 'web-stories')}</Label>
+      {lockPadding ? (
+        <LockedPaddingControls
+          padding={padding}
+          pushUpdateForObject={pushUpdateForObject}
+        />
+      ) : (
+        <UnlockedPaddingControls
+          padding={padding}
+          pushUpdateForObject={pushUpdateForObject}
+        />
+      )}
+    </Row>
+  );
+}
+
+PaddingControls.propTypes = {
```

Comment:
I would highly recommend splitting this file into small files to contain the 3 components (`PaddingControls`, `LockedPaddingControls `, and `UnlockedPaddingControls`) declared here.

Reply:
Done.